demos/python_demos/asl_recognition_demo/asl_recognition_demo/video_stream.py

```diff
@@ -0,0 +1,154 @@
+            raise Exception('File not found: {}'.format(path))
+
+        self._trg_fps = trg_fps
+        assert self._trg_fps > 0
+        self._batch_size = batch_size
+        assert self._batch_size > 0
+
+        self._cap = cv2.VideoCapture(path)
+        assert self._cap.isOpened(), "Can't open " + path
+        self._cap.set(cv2.CAP_PROP_BUFFERSIZE, 1)
+
+        self._image_shape = [int(self._cap.get(cv2.CAP_PROP_FRAME_HEIGHT)),
+                             int(self._cap.get(cv2.CAP_PROP_FRAME_WIDTH)),
+                             3]
+        self._batch_shape = [batch_size] + self._image_shape
+
+        self._image_buffer_size = int(np.prod(self._image_shape))
+        self._batch_buffer_size = int(np.prod(self._batch_shape))
+
+        self._raw_frame = Array('B', self._image_buffer_size, lock=True)
+        self._slow_frame = Array('B', self._image_buffer_size, lock=True)
+        self._slow_batch = Array('B', self._batch_buffer_size, lock=True)
+
+    def get_live_frame(self):
+        """Returns last live frame from the input stream"""
+
+        with self._raw_frame.get_lock():
+            buffer = np.frombuffer(self._raw_frame.get_obj(), dtype=np.uint8)
+            frame = np.copy(buffer.reshape(self._image_shape))
+        return frame
+
+    def get_frame(self):
+        """Returns last frame with constant framerate from the input stream"""
+
+        with self._slow_frame.get_lock():
+            buffer = np.frombuffer(self._slow_frame.get_obj(), dtype=np.uint8)
+            frame = np.copy(buffer.reshape(self._image_shape))
+        return frame
+
+    def get_batch(self):
+        """Returns last batch of frames with constant framerate from the input stream"""
+
+        with self._slow_batch.get_lock():
+            buffer = np.frombuffer(self._slow_batch.get_obj(), dtype=np.uint8)
+            batch = np.copy(buffer.reshape(self._batch_shape))
+        return batch
+
+    def start(self):
+        """Starts internal threads"""
+
+        self._frame_generator_process = \
+            Process(target=self._frame_generator,
+                    args=(self._cap, self._raw_frame, self._image_shape))
+        self._frame_generator_process.daemon = True
+        self._frame_generator_process.start()
+
+
+        self._prodcuer_process = \
+            Process(target=self._producer,
+                    args=(self._raw_frame, self._slow_frame, self._slow_batch,
+                          self._trg_fps, self._batch_size, self._image_shape))
+        self._prodcuer_process.daemon = True
+        self._prodcuer_process.start()
+
+    def release(self):
+        """Release internal threads"""
+
+        self._frame_generator_process.terminate()
+        self._prodcuer_process.terminate()
+        self._cap.release()
+
+    @staticmethod
+    def _frame_generator(input_stream, out_frame, frame_shape):
+        """Produces live frames from the input stream"""
+
+        signal.signal(signal.SIGINT, signal.SIG_IGN)
```

Comment:
Why?

Reply:
Removed.

Reply:
`import signal` should also be removed

Reply:
Please remove the `signal` import as well.

Reply:
Done.